### PR TITLE
Prepare 5.0.0-beta3

### PR DESCRIPTION
**Pull request by Nurovek**
- Bootstrap v5.0.0-beta3 release: https://github.com/twbs/bootstrap/releases/tag/v5.0.0-beta3
- Bootstrap blog post: https://blog.getbootstrap.com/2021/03/23/bootstrap-5-beta-3/

---

Preview : https://deploy-preview-650--boosted.netlify.app/

---

- [x] Run linters;
- [ ] Run compilers;
- [x] Run tests;
- [ ] Check documentation site: examples and contents;
- [ ] Test cross-browser compatibility locally and with [BrowserStack](https://www.browserstack.com/):
  - Firefox ESR
  - Latest Edge, Chrome, Firefox, Safari
  - iOS Safari
  - Chrome & Firefox on Android
- [ ] Including RTL mode;
- [x] [sync with Bootstrap](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Syncing-with-Bootstrap)'s release and probably wait for it;
- [x] `npm run release-version $current_version $next_version` to bump version number
  - ~~if bumping a minor or major version~~ → still v5.0
- [x] `npm run release` to compile dist, update SRI hashes in doc and package the release
- [x] Prepare changelog:
  - install [Conventionnal Changelog](https://github.com/conventional-changelog/conventional-changelog) and `conventional-changelog-cli` globally
  - run `conventional-changelog -p angular -i CHANGELOG.md -s`
  - Ship list: #585
- [x] commit and push `dist` with a `chore(release)` commit message
- [ ] merge on `v5-dev`
- [ ] tag your version, and push your tag
- [ ] [create a GitHub release](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/releases/new):
  - attach zip file
  - paste CHANGELOG or Ship List in the release's description
- [ ] We don't publish v5.0 docs yet
- [ ] `npm pack` then `npm publish`
  - if you are already logged in NPM (with a personnal account, for example), [you'd better use a repository scoped `.npmrc` file](https://stackoverflow.com/questions/30114166/how-to-have-multiple-npm-users-set-up-locally)
  - if you're releasing a pre-release, use `--tag`, eg for v5-alpha1 `npm publish boosted-5.0.0-alpha1.tgz --tag next`
- [ ] check release on [NPM](https://www.npmjs.com/package/boosted), [Packagist](https://packagist.org/packages/orange-opensource/orange-boosted-bootstrap)…
- [ ] make an announcement in Plazza :tada:

### CODE REVIEW

File: scss/_accordion.scss [hunk 1]
```diff
@@ -58,39 +57,40 @@
 }
 
 .accordion-item {
+  margin-bottom: -$accordion-border-width;
```
Comment (Lausselloic) [inline scss/_accordion.scss:60]: following lines change the look and feel of our accordions 
https://deploy-preview-650--boosted.netlify.app/docs/5.0/components/accordion/
https://v5-dev--boosted.netlify.app/docs/5.0/components/accordion/

File: scss/_list-group.scss [hunk 2]
```diff
@@ -12,6 +12,17 @@
   @include border-radius($list-group-border-radius);
 }
 
+.list-group-numbered {
```
Comment (Lausselloic) [inline scss/_list-group.scss:15]: Need to get feedback from brand for numbered list-group https://deploy-preview-650--boosted.netlify.app/docs/5.0/components/list-group/#numbered

File: scss/_variables.scss [hunk 3]
```diff
@@ -1240,10 +1240,11 @@ $card-footer-color:                 $gray-700 !default; // Boosted mod
 $accordion-padding-y:                     $spacer / 2 !default;
 $accordion-padding-x:                     $spacer / 2 !default;
 $accordion-color:                         $body-color !default;
-$accordion-bg:                            transparent !default;
+$accordion-bg:                            $body-bg !default;
 $accordion-border-width:                  $border-width / 2 !default;
 $accordion-border-color:                  $gray-500 !default;
 $accordion-border-radius:                 $border-radius !default;
+$accordion-inner-border-radius:           subtract($accordion-border-radius, $accordion-border-width) !default;
```
Comment (Lausselloic) [inline scss/_variables.scss:1247]: No radius in Orange brand put it to 0

File: site/content/docs/5.0/examples/headers/index.html [hunk 4]
```diff
@@ -0,0 +1,290 @@
+---
```
Comment (Lausselloic) [inline site/content/docs/5.0/examples/headers/index.html:1]: need to custom this sample with Orange design

File: site/content/docs/5.0/examples/heroes/index.html [hunk 5]
```diff
@@ -0,0 +1,123 @@
+---
```
Comment (Lausselloic) [inline site/content/docs/5.0/examples/heroes/index.html:1]: need to suctom heroes sample to Orange Brand

File: site/content/docs/5.0/examples/jumbotron/index.html [hunk 6]
```diff
@@ -0,0 +1,45 @@
+---
```
Comment (Lausselloic) [inline site/content/docs/5.0/examples/jumbotron/index.html:1]: need to brand align

File: site/content/docs/5.0/examples/pricing/index.html [hunk 7]
```diff
@@ -6,77 +6,145 @@
...
-  <p class="h5 my-0 me-md-auto fw-normal">Company name</p>
-  <nav class="my-2 my-md-0 me-md-3">
-    <a class="p-2 text-dark" href="#">Features</a>
-    <a class="p-2 text-dark" href="#">Enterprise</a>
-    <a class="p-2 text-dark" href="#">Support</a>
-    <a class="p-2 text-dark" href="#">Pricing</a>
-  </nav>
-  <a class="btn btn-primary" href="#">Sign up</a>
-</header>
+<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
```
Comment (Lausselloic) [inline site/content/docs/5.0/examples/pricing/index.html:9]: need to check logo

File: site/content/docs/5.0/examples/pricing/pricing.css [hunk 8]
```diff
@@ -1,3 +1,7 @@
+body {
+  background-image: linear-gradient(180deg, #eee, #fff 100px, #fff);
```
Comment (Lausselloic) [inline site/content/docs/5.0/examples/pricing/pricing.css:2]: check orange design

File: site/content/docs/5.0/examples/sidebars/index.html [hunk 9]
```diff
@@ -0,0 +1,387 @@
+---
```
Comment (Lausselloic) [inline site/content/docs/5.0/examples/sidebars/index.html:1]: need to check orange brand

File: site/content/docs/5.0/examples/sign-in/index.html [hunk 10]
```diff
@@ -11,10 +11,16 @@
   <form>
     <img class="mb-4" src="/docs/{{< param docs_version >}}/assets/brand/orange-logo.svg" alt="" width="50" height="50">
     <h1 class="h3 mb-3 fw-normal">Please sign in</h1>
-    <label for="inputEmail" class="visually-hidden">Email address</label>
-    <input type="email" id="inputEmail" class="form-control" placeholder="Email address" required autofocus>
-    <label for="inputPassword" class="visually-hidden">Password</label>
-    <input type="password" id="inputPassword" class="form-control" placeholder="Password" required>
+
```
Comment (Lausselloic) [inline site/content/docs/5.0/examples/sign-in/index.html:14]: need to review the form display

File: site/content/docs/5.0/examples/starter-template/index.html [hunk 11]
```diff
@@ -5,46 +5,46 @@
...
 ---
 
-<nav class="navbar navbar-expand-md navbar-dark bg-dark fixed-top">
-  <div class="container-fluid">
-    <a class="navbar-brand" href="#">Navbar</a>
-    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarsExampleDefault" aria-controls="navbarsExampleDefault" aria-expanded="false" aria-label="Toggle navigation">
-      <span class="navbar-toggler-icon"></span>
-    </button>
+<div class="col-lg-8 mx-auto p-3 py-md-5">
+  <header class="d-flex align-items-center pb-3 mb-5 border-bottom">
```
Comment (Lausselloic) [inline site/content/docs/5.0/examples/starter-template/index.html:9]: need to check logo

**Review by Lausselloic — CHANGES_REQUESTED** [review]
- some sample need brand alignement. Maybe fix the design on accordions and create other PR for Orange branded samples
- no need to add dist files in this MR only for release, not sure if this PR is intended to be the release or just the bootstrap merge
- What's that new dist/css/boosted-rtl.rtl.css.map? Is there a merge side effect that produce new files?
- before release need to change the boosted version in _config.yml but not at time (if this PR is not for release)

File: site/content/docs/5.0/examples/starter-template/index.html (hunk 11 above)
Comment (Nurovek) [reply site/content/docs/5.0/examples/starter-template/index.html:9]: > * some sample need brand alignement. Maybe fix the design on accordions and create other PR for Orange branded samples
> * no need to add dist files in this MR only for release, not sure if this PR is intended to be the release or just the bootstrap merge
> * What's that new dist/css/boosted-rtl.rtl.css.map? Is there a merge side effect that produce new files?
> * before release need to change the boosted version in _config.yml but not at time (if this PR is not for release)

It's meant to be the release of Boosted beta3

**Comment by julien-deramond** [comment]
The commit https://github.com/twbs/bootstrap/pull/33401/files is listed in https://github.com/twbs/bootstrap/releases/tag/v5.0.0-beta3 but I don't see this modification in this PR.

Am I wrong or we could use `https://img.shields.io/github/workflow/status/Orange-OpenSource/Orange-Boosted-Bootstrap/JS%20Tests/v5-dev?label=JS%20Tests&logo=github` in our `README.md` too?

And when V5 will be the `main` branch of Boosted, modify it in `https://img.shields.io/github/workflow/status/Orange-OpenSource/Orange-Boosted-Bootstrap/JS%20Tests/main?label=JS%20Tests&logo=github`.

**Comment by julien-deramond** [comment]
`build/ship.sh` has been deleted in Bootstrap via https://github.com/twbs/bootstrap/pull/33326/files and it is not deleted in this PR.
It is mentioned in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/DoD-for-release but shouldn't it be deleted as well and possibly copy those commands in the Wiki?
Or if this script is useful, maybe rename it so that it is clear that it belongs only to Boosted?

File: CODE_OF_CONDUCT.md [hunk 12]
```diff
@@ -34,7 +34,7 @@ This Code of Conduct applies both within project spaces and in public spaces whe
 
 ## Enforcement
 
-Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at mdo@getbootstrap.com. The project team will review and investigate all complaints, and will respond in a way that it deems appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at kevin.lediouron.ext@orange.com. The project team will review and investigate all complaints, and will respond in a way that it deems appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
```
[the remark this answers is not in the document]
Comment (ffoodd) [reply CODE_OF_CONDUCT.md:37]: Or leave it as-is :)

**Comment by ffoodd** [comment]
My two cents:
* drop the `build/ship.sh`, it basically only copies the docs out of the repo to ease switching ti `gh-pages` branch. Can be done manually.
* example customization to apply Orange brand is tracked in a specific issue, no hurry on those.
* list group is not tracked by the brand as they don't use it: keep it as-is, without asking them  (if it's not too late :D)
* check accordions through Percy, it's crying already… and make it pass!

**Comment by julien-deramond** [comment]
This is my 1st Boosted code review, so take my comments cautiously 😇 

- [4c7a3e8adf](https://github.com/twbs/bootstrap/commit/4c7a3e8adf73bf55a5fba5905b356468be17bc0e) adds Sass docs  (variables, mixins, and loops) to most pages. Some of them specific to Boosted aren't they missing? I'm thinking about `@each $name in map-keys($btn-social-networks) {` in `_button.scss` for example.
- [d4e6b973](https://github.com/twbs/bootstrap/commit/d4e6b9731b0671d3f5f7a18c12a70844207e8d03) This PR doesn't modify `config.yml`:
  - Still references `beta2` instead of `beta3`
  - `popper` and `popper_hash` are not modified for example
- I'm sad to see "List group" disappear. Can't be keep it? It must be used by projects in 4.6.
- The problems with the Accordions has already been mentionned by Loïc

File: scss/_accordion.scss (hunk 1 above)
Comment (Nurovek) [reply scss/_accordion.scss]: Accordion has been re-accorded to Orange's look & feel (b547626)

File: scss/_variables.scss (hunk 3 above)
Comment (ffoodd) [reply scss/_variables.scss:1247]: If `$border-radius` is `null`, no need to override this: it won't output anything.

File: scss/_accordion.scss (hunk 1 above)
Comment (Lausselloic) [reply scss/_accordion.scss]: ok

File: scss/_accordion.scss [hunk 13]
```diff
@@ -58,39 +56,36 @@
 }
 
 .accordion-item {
+
```
Comment (Lausselloic) [inline scss/_accordion.scss:59]: need to comment bootstrap initial values and add a comment to explain its a boosted mod : 
Bootstrap initial values : 
```
margin-bottom: -$accordion-border-width;
  background-color: $accordion-bg;
  border: $accordion-border-width solid $accordion-border-color;
```

File: scss/_accordion.scss [hunk 14]
```diff
@@ -58,39 +56,36 @@
...
   &:first-of-type {
+
```
Comment (Lausselloic) [inline scss/_accordion.scss:61]: Keep bootstrap mixin include, rounded is disabled on boosted so no effect
```
@include border-top-radius($accordion-border-radius);
```

File: scss/_accordion.scss [hunk 15]
```diff
@@ -58,39 +56,36 @@
...
     .accordion-button {
       @include border-top-radius($accordion-border-radius);
```
Comment (Lausselloic) [inline scss/_accordion.scss:63]: wrong variable value according to bootstrap need to be `accordion-inner-border-radius`and so lint won't cry for unusued variable

File: scss/_accordion.scss [hunk 16]
```diff
@@ -58,39 +56,36 @@
...
     }
   }
 
+  // Only set a border-radius on the last item if the accordion is collapsed
   &:last-of-type {
```
Comment (Lausselloic) [inline scss/_accordion.scss:68]: keep the `margin-bottom: 0` will have no impact on our design and keep align with bootstrap

File: scss/_accordion.scss [hunk 17]
```diff
@@ -58,39 +56,36 @@
...
+    @include border-bottom-radius($accordion-border-radius);
     border-bottom: $accordion-border-width solid $accordion-border-color; // Boosted mod: borders are set on .accordion-header
 
     .accordion-button {
       // Only set a border-radius on the last item if the accordion is collapsed
       &.collapsed {
-        // Boosted mod: borders are set on .accordion-header
         @include border-bottom-radius($accordion-border-radius);
```
Comment (Lausselloic) [inline scss/_accordion.scss:75]: wrong variable value according to bootstrap need to be `accordion-inner-border-radius`and so lint won't cry for unusued variable

File: site/data/sidebar.yml [hunk 18]
```diff
@@ -68,7 +68,6 @@
     - title: Close button
     - title: Collapse
     - title: Dropdowns
-    - title: List group
```
Comment (Lausselloic) [inline site/data/sidebar.yml:71]: Keep list-group link into the menu

**Review by Lausselloic — CHANGES_REQUESTED** [review]

File: site/content/docs/5.0/examples/starter-template/index.html (hunk 11 above)
Comment (Lausselloic) [reply site/content/docs/5.0/examples/starter-template/index.html:9]: will be adressed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/380

File: site/content/docs/5.0/examples/sign-in/index.html (hunk 10 above)
Comment (Lausselloic) [reply site/content/docs/5.0/examples/sign-in/index.html:14]: will be adressed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/380

File: site/content/docs/5.0/examples/sidebars/index.html (hunk 9 above)
Comment (Lausselloic) [reply site/content/docs/5.0/examples/sidebars/index.html:1]: will be adressed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/380

File: site/content/docs/5.0/examples/pricing/pricing.css (hunk 8 above)
Comment (Lausselloic) [reply site/content/docs/5.0/examples/pricing/pricing.css:2]: will be adressed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/380

File: site/content/docs/5.0/examples/pricing/index.html (hunk 7 above)
Comment (Lausselloic) [reply site/content/docs/5.0/examples/pricing/index.html:9]: will be adressed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/380

File: site/content/docs/5.0/examples/headers/index.html (hunk 4 above)
Comment (Lausselloic) [reply site/content/docs/5.0/examples/headers/index.html:1]: will be adressed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/380

File: scss/_list-group.scss (hunk 2 above)
Comment (Lausselloic) [reply scss/_list-group.scss:15]: will see later

File: site/content/docs/5.0/examples/jumbotron/index.html (hunk 6 above)
Comment (Lausselloic) [reply site/content/docs/5.0/examples/jumbotron/index.html:1]: will be adressed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/380

File: site/content/docs/5.0/examples/heroes/index.html (hunk 5 above)
Comment (Lausselloic) [reply site/content/docs/5.0/examples/heroes/index.html:1]: will be adressed in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/380

**Review by ffoodd — CHANGES_REQUESTED** [review]
Regarding Pa11y, would you mind re-running the action? Some paged time out.
New errors all came up with Bootstrap's new examples, I'll try to fix them upstream—you may want to ignore them for now.

And Percy only shows a single change, where it was previously missing a snapshot. IMHO you're safe to approve the build :)

**Comment by ffoodd** [comment]
Pa11y errors should be covered in [Bootstrap #33772](https://github.com/twbs/bootstrap/pull/33772), please check its status.

Some won't so you'll have to ignore them the most specifically you can (e.g. maybe try to disable a test only on specific URLs? Not sure it is possible) — or you'll need to override Boosted's example to fix examples in this repo.

**Review by Lausselloic — APPROVED** [review]

File: config.yml [hunk 19]
```diff
@@ -53,7 +53,7 @@ params:
   social_image_path:    /docs/5.0/assets/brand/orange-social.png
   social_logo_path:     /docs/5.0/assets/brand/orange-social-logo.png
 
-  current_version:      "5.0.0-beta2"
+  current_version:      "5.0.0-beta3"
   current_ruby_version: "5.0.0.beta2"
```
Comment (Lausselloic) [inline config.yml:57]: ruby will stay in beta2?

Comment (Nurovek) [reply config.yml:57]: Mistake I guess. Modifying it right now

File: config.yml [hunk 20]
```diff
@@ -63,23 +63,23 @@ params:
...
-    helvetica_rtl:      "https://cdn.jsdelivr.net/npm/boosted@5.0.0-beta2/dist/css/orange-helvetica.rtl.min.css"
+    helvetica_rtl:      "https://cdn.jsdelivr.net/npm/boosted@5.0.0-beta3/dist/css/orange-helvetica.rtl.min.css"
     helvetica_rtl_hash: "sha384-2N+Axfck8AYLrpdsZD0N1m7VB20fwJPh1SzLbOhtlxThb8GEzCaX4uJwMHLxCyxh"
-    js:                 "https://cdn.jsdelivr.net/npm/boosted@5.0.0-beta2/dist/js/boosted.min.js"
+    js:                 "https://cdn.jsdelivr.net/npm/boosted@5.0.0-beta3/dist/js/boosted.min.js"
     js_hash:            "sha384-VzJG8txEnR93rWKSsWA5q4qsd8HzVgTF9D8j96AfQRwWr9a5M6IFgeJnd4Sbpg4P"
-    js_bundle:          "https://cdn.jsdelivr.net/npm/boosted@5.0.0-beta2/dist/js/boosted.bundle.min.js"
+    js_bundle:          "https://cdn.jsdelivr.net/npm/boosted@5.0.0-beta3/dist/js/boosted.bundle.min.js"
     js_bundle_hash:     "sha384-Vc9TemN3pAGe1Yu1xsF6n3b8t65sxppt3SP/znqPkSmcNdTD9kKa3XqdlegP0jqP"
     popper:             "https://cdn.jsdelivr.net/npm/@popperjs/core@2.8.2/dist/umd/popper.min.js"
```
Comment (Lausselloic) [inline config.yml:84]: Strange if this is a merge from bootstrap beta3 tag, poper need to be updated https://github.com/twbs/bootstrap/blob/v5.0.0-beta3/config.yml#L78 and his hash too

Comment (Nurovek) [reply config.yml:84]: Modified manually

**Comment by Lausselloic** [comment]
From where are comming new files `*-rtl.rtl*`? They weren't in the repo before that merge

**Comment by Nurovek** [comment]
> From where are comming new files `*-rtl.rtl*`? They weren't in the repo before that merge

It seems that is the case in bootstrap too https://github.com/twbs/bootstrap/pull/33439/files

**Comment by ffoodd** [comment]
Nope, there's no such files (look closer at `*-rtl.rtl`). There's no `boosted-rtl.scss` source so you shouldn't have `boosted-rtl.css` : my bet is some v4 leftovers when switching to v5.

Maybe add thos files to `.gitignore`? But for now you need to remove them from this PR :)

**Comment by Lausselloic** [comment]
added file removed and dist regenrated with my last commit

**Pull request merged**